client/src/modules/transaction-type/transaction-type.ctrl.js

```diff
@@ -96,25 +84,32 @@ function TransactionTypeController(TransactionType, TransactionTypeStore, Notify
       .then(res => {
         if (!res) { return; }
         Notify.success('FORM.INFO.UPDATE_SUCCESS');
-        load();
+        loadTransactionTypes();
       })
       .catch(Notify.handleError);
   }
 
-  function load() {
+  function loadTransactionTypes() {
     TransactionType.read()
       .then(list => {
-        vm.gridOptions.data = list;
-
-        list.forEach(type => {
-          type._popover = type.fixed ? $translate.instant('TRANSACTION_TYPE.FIXED_INFO') : '';
-        });
+        vm.gridOptions.data = list.map(assignTransactionTypeLabels);
       })
       .catch(Notify.handleError);
   }
 
+  // Assign translatable labels to each transaction type based on the hardcoded
+  // database strings
+  function assignTransactionTypeLabels(transactionType) {
+    // @TODO(sfount) The `transaction_type` database currently hard codes 'income',
+    //               'expense' and 'other' transaction types. When this is updated
+    //               with a more data driven approach it should include translatable
+    //               labels
```

Comment:
Good comment!  Mind making an issue about it?  I can't recall if one exists or not..